packages/svelte2tsx/src/svelte2tsx/processInstanceScriptContent.ts

```diff
@@ -385,3 +394,33 @@ export function processInstanceScriptContent(
         generics
     };
 }
+
+function transformInterfacesToTypes(tsAst: ts.SourceFile, str: MagicString, astOffset: any) {
+    tsAst.statements.filter(ts.isInterfaceDeclaration).forEach((node) => {
+        str.overwrite(
+            node.getStart() + astOffset,
+            node.getStart() + astOffset + 'interface'.length,
+            'type'
+        );
+
+        if (node.heritageClauses?.length) {
+            const extendsStart = str.original.indexOf('extends', node.getStart() + astOffset);
+            str.overwrite(extendsStart, extendsStart + 'extends'.length, '=');
+
+            const extendsList = node.heritageClauses[0].types;
+            let prev = extendsList[0];
+            extendsList.slice(1).forEach((heritageClause) => {
+                str.overwrite(
+                    prev.getEnd() + astOffset,
+                    heritageClause.getStart() + astOffset,
+                    ' & '
```

Comment:
There's still [interface merging](https://www.typescriptlang.org/docs/handbook/declaration-merging.html#merging-interfaces). But this seems like an edge case though. Don't really know if anyone uses this feature in a svelte file.

Reply:
I hope noone does, but even if they do, this only affects the dts output, which is worse than it would be with this change.